src/System.Windows.Forms/src/System/Windows/Forms/DataObject.cs

```
@@ -1414,6 +1416,10 @@ private object GetDataFromOleHGLOBAL(string format, out bool done)
                     catch
```

should we have these catch-all blocks?

---

Unfortunately its required here for two reasons:

* GetDataFromHGLOBAL reports failure via exception, but you don't know which exception it is . It does complex dispatching and some branches can execute arbitrary user code while deserializing an serialized object (not for the builtin formats, but custom formats still support arbitrary serialized objects)
* For the interop calls, while you could be more specific, you really want to catch all HRESULT-based exceptions, but some HRESULT-based exceptions get translated. So being specific requires a lot of research to cover exactly the set of HRESULT-exceptions that can get translated.

If you want to improves this you have to
* move the deserialization try-catch to where its really needed (i.e. where the deserialization happens)
* make it TryGetDataFromHGLOBAL so it can report failures without throwing arbitrary exceptions
* stop using IDataObject from corefx, make your own declaration which returns HRESULT explicitely so it doesn't get translated when you don't want it to

Certainly not something I'd want to be doing in a bugfix PR. If you think its worth to improve we should create a separate issue.